### PR TITLE
Umer Completed Changes RB

### PR DESCRIPTION
**Pull request by ufarooq82**

### CODE REVIEW

File: src/test/resources/features/medicals/Validity_Medical_Certificates.feature [hunk 1]
```diff
@@ -15,49 +15,62 @@ Feature:Certificate expiry
...
   Scenario Outline:Medical pilots license expiry rule, must renew their licenses
-
     Given Mike is a <Class> pilot
-    And   the following personal details:
+    And   his personal details are as followings:
           | Age      | For          |
           | <Age>    | <For>        |
-    When  Mike license <Duration>
-    Then  his license status should be dropped to following levels:
+    When  his last medical was on <Last Medical>
```
Comment (ufarooq82) [inline src/test/resources/features/medicals/Validity_Medical_Certificates.feature]: Need improvement

File: src/test/resources/features/medicals/Validity_Medical_Certificates.feature [hunk 2]
```diff
@@ -207,84 +90,25 @@ Feature:Certificate expiry
...
-#    Unable to fly if not renewed
-#
-#      | Class  | DOB        | Issue Date       | Expires     | Current Date | Eligible or Not Eligible | Not Dropped or Drops Once Expired                  |
-#      | Second | 15-01-1960 | 15-01-2007       | 15-12-2009  | 14-12-2009   |   Eligible               | Not Dropped                                        |
-#      | Second | 15-01-1960 | 15-01-2007       | 15-12-2009  | 15-01-2010   |   Not Eligible           | Unable to fly                                      |
+        | Class | Age      | Pilot Category          | Last Medical  |  Next Level   | Next Medical  |
+        | Third | Over 40  | Private / Recreational  | 24 months     |  Level 5      | Unable to fly |
+        | Third | Over 40  | Private / Recreational  | 20 months     |  No           |   No          |
+        | Third | Over 40  | Private / Recreational  | 10 months     |  No           |   No          |
+        | Third | Over 40  | Private / Recreational  | 1 month       |  No           |   No          |
```
Comment (globalworming) [inline src/test/resources/features/medicals/Validity_Medical_Certificates.feature]: i am not 100% comfortable with the columns. 
"Next Level: No" or "Next Medical: Unable to fly" doesn't sound nice to me. There must be something better sounding...

"Current Level: unchanged"? 
"Next Medical: no next medical"
though  `And his next medical should be no later than <no next medical>` also sounds weird. 

maybe next medical should always say when.. even if it is unchanged?

```suggestion
        | Class | Age      | Pilot Category          | Last Medical  |  Level   | Next Medical  |
        | Third | Over 40  | Private / Recreational  | 24 months     |  5      | - |
        | Third | Over 40  | Private / Recreational  | 20 months     |  unchanged           |   4         |
        | Third | Over 40  | Private / Recreational  | 10 months     |  unchanged           |   14          |
        | Third | Over 40  | Private / Recreational  | 1 month       |  unchanged           |   23          |
```

something like that? :)

**Comment by ufarooq82** [comment]
Hi @globalworming ,

Thanks for feedback. Made changes as requested and commit the code but seems not showing in here ? thanks

image

**Comment by ufarooq82** [comment]
still showing last week commits 
image

File: @First-Class-Pilots-Rules-Journey-Scenarios.png
Comment (globalworming) [inline @First-Class-Pilots-Rules-Journey-Scenarios.png:1]: images may be better in an img directory
and not sure if `@` in name is a good idea

File: README.md [hunk 3]
```diff
@@ -31,6 +31,246 @@ All of the FAA validities are measured from the last day of the month.
 
 ![image](medicals.png)
 
+# Articles:-
+https://johnfergusonsmart.com/feature-mapping-a-lightweight-requirements-discovery-practice-for-agile-teams/
+https://johnfergusonsmart.com/feature-mapping-a-simpler-path-from-stories-to-executable-acceptance-criteria/
+
+# You can see the big picture in the following diagram:
+
+! [image](Bigpicture.png)
```
Comment (globalworming) [inline README.md:40]:
```suggestion
![image](Bigpicture.png)
```

File: README.md [hunk 4]
```diff
@@ -31,6 +31,246 @@ All of the FAA validities are measured from the last day of the month.
...
+Feature Mapping works best as a collaborative activity. Many teams use the classic Three-Amigos trio:
+
+A BA or product owner, representing business knowledge;
+A developer, representing the technical perspective;
+And a tester, to make sure the requirements are testable.
+
+BAs might write some initial acceptance criteria for the story beforehand, and that’s great. 
+But these acceptance criteria can simply be one-liner business-rules, and not fully-blown Given-When-Then statements.
+
+# 1) -------------------> We start with the story
```
Comment (globalworming) [inline README.md:55]:
```suggestion
# 1) We start with the story
```

File: README.md [hunk 5]
```diff
@@ -31,6 +31,246 @@ All of the FAA validities are measured from the last day of the month.
...
+
+(![](Start with Story.png)
```
Comment (globalworming) [inline README.md:57]:
```suggestion
(![//FIXME](Start with Story.png)
```

File: README.md [hunk 6]
```diff
@@ -31,6 +31,246 @@ All of the FAA validities are measured from the last day of the month.
...
+
+A BA or product owner, representing business knowledge;
+A developer, representing the technical perspective;
+And a tester, to make sure the requirements are testable.
+
+BAs might write some initial acceptance criteria for the story beforehand, and that’s great. 
+But these acceptance criteria can simply be one-liner business-rules, and not fully-blown Given-When-Then statements.
+
+# 1) -------------------> We start with the story
+
```
Comment (globalworming) [inline README.md:56]: add link to mural maybe?

File: README.md [hunk 7]
```diff
@@ -31,6 +31,246 @@ All of the FAA validities are measured from the last day of the month.
...
+# a) Please check  README.MD file for Figure for Representing negative scenarios@Third-Class-Pilots-Rules-Journey-Scenarios
+
+!image [] (![](Representing negative scenarios@Third-Class-Pilots-Rules-Journey-Scenarios.png))
+
+# b) Please check  README.MD file for Figure for Representing negative scenarios@Third-Class-Pilots-Rules-Journey-Scenarios
+
+!image [] (![](Representing negative scenarios two@Third-Class-Pilots-Rules-Journey-Scenarios.png))
+
+
 ## The Exercise
```
Comment (globalworming) [inline README.md:274]: check below if relevant

File: README.md [hunk 8]
```diff
@@ -31,6 +31,246 @@ All of the FAA validities are measured from the last day of the month.
...
+
+!image [](![](NegativeCases@Second-Class-Pilots-Rules-Journey-Scenarios.png))
+
+# c) NegativeCases@Third-Class-Pilots-Rules-Journey-Scenarios
+
+!image [](![](NegativeCases@Third-Class-Pilots-Rules-Journey-Scenarios.png))
+
+# 9.1) ------------#Counter-examples complete the picture ^^^^-------------------------------------------------
+
+# 9.2) --------------Let’s look at another requirement-------------------------------------------------------
```
Comment (globalworming) [inline README.md:250]: dont :)

File: README.md [hunk 9]
```diff
@@ -31,6 +31,246 @@ All of the FAA validities are measured from the last day of the month.
...
+
+However, not all negative scenarios are useful. Field validation rules are fine if the rules relate to business requirements. 
+
+But a scenario checking for badly-formatted dates or numbers would normally be reserved for unit testing.
+
+For example, the rule that a teacher can return an essay if there is a mark of 6 or under begs the question: can a teacher return an essay if the marks are good? 
+
+This would justify a separate scenario:
+
+Tess records 9/9/9 for Stu's essay and tries to return it, but she is not allowed to.
```
Comment (globalworming) [inline README.md:232]: I'd remove johns stuff from the article

**Pull request merged**